src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -427,15 +427,28 @@ def create_snapshot(resource_group_name, snapshot_name, location=None, size_gb=N
     if size_gb is None and option == DiskCreateOption.empty:
         raise CLIError('Please supply size for the snapshots')
 
-    snapshot = Snapshot(location, disk_size_gb=size_gb, creation_data=creation_data,
-                        account_type=sku, tags=(tags or {}))
+    snapshot = Snapshot(location, creation_data, (tags or {}), _get_sku_object(sku), disk_size_gb=size_gb)
     client = _compute_client_factory()
     return client.snapshots.create_or_update(resource_group_name, snapshot_name, snapshot)
 
 
+def _get_sku_object(sku):
+    if supported_api_version(ResourceType.MGMT_COMPUTE, min_api='2017-03-30'):
+        DiskSku = get_sdk(ResourceType.MGMT_COMPUTE, 'DiskSku', mod='models')
+        return DiskSku(sku)
+    return sku
+
+
+def _set_sku(instance, sku):
+    if supported_api_version(ResourceType.MGMT_COMPUTE, min_api='2017-03-30'):
+        instance.sku = get_sdk(ResourceType.MGMT_COMPUTE, 'DiskSku', mod='models')(sku)
+    else:
+        instance.account_type = sku
```

Comment:
Have you tested this by switching to the other profile to verify this all works as expected?